workers/email_reports/rpm_decrease/data.go

```diff
@@ -2,71 +2,52 @@ package rpm_decrease
 	"bytes"
-	"encoding/json"
 	"fmt"
 	"github.com/m6yf/bcwork/modules"
-	"github.com/m6yf/bcwork/modules/compass"
-	"github.com/m6yf/bcwork/utils/helpers"
 	"github.com/m6yf/bcwork/workers/email_reports"
 	"github.com/rs/zerolog/log"
-	"sort"
 	"strings"
 	"text/template"
 	"time"
 )
 
-const (
-	HourlyPercentage          = 0.4
-	RPMPubImpsThreshold int64 = 50
-)
-
-func computeAverage(aggregated map[string][]email_reports.AggregatedReport, worker *Worker) map[string][]AlertsEmails {
-	amDomainData := make(map[string][]email_reports.AggregatedReport)
-
-	lastCompleteHour := time.Now().In(email_reports.Location).Truncate(time.Hour)
-	yesterdayHour := lastCompleteHour.Add(-24 * time.Hour)
-	sevenDaysAgoHour := lastCompleteHour.Add(-7 * 24 * time.Hour)
-
-	lastCompleteHourUnix := lastCompleteHour.Unix() / 100
-	yesterdayHourUnix := yesterdayHour.Unix() / 100
-	sevenDaysAgoHourUnix := sevenDaysAgoHour.Unix() / 100
-
-	for key, aggs := range aggregated {
-		for _, agg := range aggs {
-			if agg.DataStamp == lastCompleteHourUnix || agg.DataStamp == yesterdayHourUnix || agg.DataStamp == sevenDaysAgoHourUnix {
-				amDomainData[key] = append(amDomainData[key], agg)
-			}
-		}
-	}
-
+func compareResults(amDomainData map[string][]email_reports.AggregatedReport, percentage float64, userData map[string]string) map[string][]AlertsEmails {
+	alerts := make(map[string]email_reports.AggregatedReport)
 	repo := AlertsEmails{}
-	emailReports := compareResults(amDomainData, repo, worker)
-
-	avgDataMap := make(map[string][]AlertsEmails)
-	for _, repo := range emailReports {
-		avgDataMap[repo.Email] = append(avgDataMap[repo.Email], repo)
-	}
-
-	return avgDataMap
-}
+	yesterday, startOfLastWeek, today := email_reports.GetDate()
 
-func compareResults(amDomainData map[string][]email_reports.AggregatedReport, repo AlertsEmails, worker *Worker) []AlertsEmails {
 	var emailReports []AlertsEmails
-
 	for key, reports := range amDomainData {
-		if len(reports) < 3 {
-			continue
+		totalYesterday := 0.0
+		totalLastWeek := 0.0
+		countYesterday := 0
+		countLastWeek := 0
+		totalToday := 0.0
+		countToday := 0
+
+		for _, report := range reports {
+			if report.DataStamp == yesterday {
+				totalYesterday += report.LoopingRatio
+				countYesterday++
+			} else if report.DataStamp == startOfLastWeek {
+				totalLastWeek += report.LoopingRatio
+				countLastWeek++
+			} else if report.DataStamp == today {
+				totalToday += report.LoopingRatio
+				countToday++
+			}
 		}
 
-		sort.Slice(reports, func(i, j int) bool {
-			return reports[i].DataStamp < reports[j].DataStamp
-		})
+		if countToday == 0 || (countYesterday == 0 && countLastWeek == 0) {
+			continue
+		}
 
-		if reports[2].RPM < HourlyPercentage*(reports[1].RPM) {
+		if totalToday < percentage*(totalYesterday+totalLastWeek) {
 			latestReport := reports[len(reports)-1]
-			//emailKey := strings.Split(key, "|")
+			alerts[key] = latestReport
+			emailKey := strings.Split(key, "|")
 			repo = AlertsEmails{
```

Comment:
No need to create temporary variable if you after creation you append it
```suggestion
		 emailReports = append(emailReports, AlertsEmails{
		 		Email:        userData[emailKey[0]],
				AM:           key,
				FirstReport:  latestReport,
				SecondReport: reports,
			}
```

Reply:
done